tests/core/test_read_manifest.py

```diff
@@ -51,11 +51,8 @@ def test_read_defaults(datadir):
         actual = json.load(f)
 
     # expected
-    with open(os.path.join(datadir, 'defaults.json'), 'r') as f:
-        expected = json.load(f)
-
-    # special case (initialized in constructor)
-    expected['design']['value'] = 'test'
+    expected_chip = siliconcompiler.Chip('test')
+    expected = expected_chip.schema.cfg
```

Comment:
This is not necessarily about your changes, but while we're touching this file maybe it's worth fixing - I think it's confusing what this test is doing at first glance. It sounds like the basic idea is to make sure a read/write preserves the manifest fully, so isn't the following much shorter test sufficient:

```python
chip = siliconcompiler.Chip('foo')
chip.write_manifest('foo.json')

chip2 = siliconcompiler.Chip('foo')
chip2.read_manifest('foo.json')

assert chip.schema.cfg == chip2.schema.cfg
```

In the existing test, reading the manifest back into the same chip object doesn't do a full read, just a merge (but I'm not sure that's what was intended). I'm also not really sure what dumping the json to a file and then reading it back into a dictionary was trying to accomplish either.

Reply:
It's possible this test needs to be reworked to test the functionality better, which I wasn't sure what the true test was aiming for. I'd like to see if there is a path for the test that doesn't require the defaults manifest, but if that is not possible, we can restore it.